pkg/registries/registries.go

```diff
@@ -155,10 +201,31 @@ func EditRegistriesConfig(config *sysregistriesv2.V2RegistriesConf, insecureRegi
 		reg.Blocked = true
 		for i := range config.Registries {
 			reg := &config.Registries[i]
-			if scopeMatchesRegistry(reg.Location, blockedReg) {
+			scope := reg.Location
+			// Set scope to Prefix if it exists
+			if reg.Prefix != "" {
+				scope = reg.Prefix
+			}
+			if regMatchesScope(scope, blockedReg) {
 				reg.Blocked = true
 			}
 		}
 	}
 	return nil
 }
+
+// IsValidRegistryOrWildcardScope returns true if scope is a valid registry or wildcard entry such as "*.registry.com"
+// This function can be used to validate the registries entries prior to calling EditRegistriesConfig
+// in the MCO or builds code
+func IsValidRegistryOrWildcardScope(scope string) bool {
+	// If scope does not contain the wildcard character, we will assume it is a regular registry entry, which is valid
+	if !strings.Contains(scope, "*") {
+		return true
+	}
+	// If it contains the wildcard character, check that it doesn't contain any invalid characters.
+	// The only valid scope would be when it has the prefix "*."
+	if strings.ContainsAny(scope[1:], "/@:*") {
```

Comment:
This doesn’t check for `scope[1]` being `.`; so is the comment incorrect, or is this intentionally not checking everything?

Reply:
Fixed